unit_tests/conftest.py

```diff
@@ -49,22 +49,29 @@ def prom_address():
     yield start_metrics_server()
 
 
-@pytest.fixture(scope='session')
-def docker_scylla():
+def setup_docker_scylla(ssl=False):
     # make sure the path to the file is base on the host path, and not as the docker internal path i.e. /sct/
     # since we are going to mount it in a DinD (docker-inside-docker) setup
     base_dir = os.environ.get("_SCT_BASE_DIR", None)
     entryfile_path = Path(base_dir) if base_dir else Path(__file__).parent.parent
-    entryfile_path = entryfile_path.joinpath('./docker/scylla-sct/entry.sh')
+    entryfile_path = entryfile_path / 'docker' / 'scylla-sct' / ('entry_ssl.sh' if ssl else 'entry.sh')
 
     alternator_flags = "--alternator-port 8000 --alternator-write-isolation=always"
     docker_version = "scylladb/scylla-nightly:5.2.0-dev-0.20220820.516089beb0b8"
     cluster = LocalScyllaClusterDummy()
+
+    if ssl:
+        curr_dir = os.getcwd()
+        os.chdir(Path(__file__).parent.parent)
+        update_certificates()
+        os.chdir(curr_dir)
```

Comment:
Not safe approach to change the dir.
If line 66 fails then we do not return back the cwd.
Would be good to have either "try-finally" block or "with %change-path%" variants here.

Reply:
Looks like the stdlib finally getting it, in 3.11:
https://github.com/python/cpython/pull/28271

After 7 years...

Reply:
Does it mean that, having py3.10, you do not plane to update this part of code?

Reply:
I intended to copy it from 3.11 code, and remove it once we upgrade to it.

meanwhile I haven't gotten back to it, since I'm not feeling too good. 

Want to take it from here ?

Reply:
I have no rights to push changes to your repo/branch. It will require creation of a new PR.
But I have created a code snippet, here it is:

```
os.chdir(curr_dir)
try:
    os.chdir(Path(__file__).parent.parent)
    update_certificates()
finally:
    os.chdir(curr_dir)
```

So, please apply it.